ddterm/shell/extension.js

```diff
@@ -140,6 +142,33 @@ class ExtensionDBusInterface {
     }
 }
 
+class DesktopEntry {
+    constructor() {
+        this.source_file = Me.dir.get_child('ddterm').get_child('com.github.amezin.ddterm.desktop');
+        this.target_file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'applications',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+    }
+
+    install() {
+        GLib.mkdir_with_parents(this.target_file.get_parent().get_path(), 0o700);
+        this.source_file.copy(this.target_file, Gio.FileCopyFlags.OVERWRITE, null, null);
+    }
+
+    uninstall() {
+        try {
+            this.target_file.delete(null);
+        } catch (e) {
+            if (!e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND))
+                throw e;
```

Comment:
Throwing here is a bad idea, because it will interrupt `disable()`.

Could you just log the error here, and not re-throw it?

And, again, if the file is not found here, it means something unexpected happened - I think it should be an error too (i. e. should be logged as an error).

Reply:
I was going to work on this next, had to take a break to do something else 😅
 
Thanks for reminding me though. Next time, if you prefer, I can request a review when I think it is worth another shot, so you don't lose too much time reviewing my WIPs.